### ISSUE TITLE
Confusion about training and validation when pretraining 

### ISSUE THREAD

**Issue by XiHuYan** [HobbitLong/SupContrast#93]
When pretraining with the crossentropy, you set a validation dataset. It's consistent with training a classification network.

However, when pretraining with the supcon loss, you used all the dataset as training set. I wonder why you didn't split them to training and validation set, just like pretraining with crossentropy? Is it because supcon will not face the risk of overfitting?

**Comment by HobbitLong** [comment]
Hi, @XiHuYan,

I think your concern is valid. We could use the same validation set (used for cross-entropy) to check supcon loss. But I suppose this way is not as informative as I validate it by checking the follow-up linear classification performance on the `val` set since this is real task we care about.

Ideally or legitimately, we should have another `test` set for reporting performance.
